expression/builtin_time.go

```diff
@@ -3020,36 +3021,82 @@ type utcTimeFunctionClass struct {
 	baseFunctionClass
 }
 
+func (c *utcTimeFunctionClass) getFlenAndDecimal4UTCTime(sc *variable.StatementContext, args []Expression) (flen, decimal int) {
+	if len(args) == 0 {
+		flen, decimal = 8, 0
+		return
+	}
+	if constant, ok := args[0].(*Constant); ok {
+		fsp, isNull, err := constant.EvalInt(nil, sc)
+		if isNull || err != nil || fsp > int64(types.MaxFsp) {
+			decimal = types.MaxFsp
+		} else if fsp < int64(types.MinFsp) {
+			decimal = types.MinFsp
+		} else {
+			decimal = int(fsp)
+		}
+	}
+	if decimal > 0 {
+		flen = 8 + 1 + decimal
+	} else {
+		flen = 8
+	}
+	return flen, decimal
+}
+
 func (c *utcTimeFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
 	if err := c.verifyArgs(args); err != nil {
 		return nil, errors.Trace(err)
 	}
-	sig := &builtinUTCTimeSig{newBaseBuiltinFunc(args, ctx)}
+	argTps := make([]evalTp, 0, 1)
+	if len(args) == 1 {
+		argTps = append(argTps, tpInt)
+	}
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpDuration, argTps...)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	bf.tp.Flen, bf.tp.Decimal = c.getFlenAndDecimal4UTCTime(bf.ctx.GetSessionVars().StmtCtx, args)
+
+	var sig builtinFunc
+	if len(args) == 1 {
+		sig = &builtinUTCTimeWithArgSig{baseDurationBuiltinFunc{bf}}
+	} else {
+		sig = &builtinUTCTimeWithoutArgSig{baseDurationBuiltinFunc{bf}}
+	}
 	return sig.setSelf(sig), nil
 }
 
-type builtinUTCTimeSig struct {
-	baseBuiltinFunc
+type builtinUTCTimeWithoutArgSig struct {
+	baseDurationBuiltinFunc
 }
 
-// eval evals a builtinUTCTimeSig.
+// evalDuration evals a builtinUTCTimeWithoutArgSig.
 // See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_utc-time
-func (b *builtinUTCTimeSig) eval(row []types.Datum) (d types.Datum, err error) {
-	const utctimeFormat = "15:04:05.000000"
-	args, err := b.evalArgs(row)
-	if err != nil {
-		return d, errors.Trace(err)
+func (b *builtinUTCTimeWithoutArgSig) evalDuration(row []types.Datum) (types.Duration, bool, error) {
+	v, err := types.ParseDuration(time.Now().UTC().Format("00:00:00"), 0)
```

Comment:
if `err` is not `nil`, the usage of `v` maybe undefined, so maybe we should:
```sql
is err != nil {
    return types.Duration{}, true, errors.Trace(err);
}
return v, false, nil
```

thus if the caller decide to ignore this error, it will get a `NULL` value, not the un-certain `v`